src/Connection/Connection.php

```diff
@@ -594,19 +576,17 @@ public function getServerVersion(): string
      *
      * @throws InvalidConfigException
      *
-     * @return Connection the currently active slave connection. `null` is returned if there is no slave available and
+     * @return self the currently active slave connection. `null` is returned if there is no slave available and
      * `$fallbackToMaster` is false.
      */
-    public function getSlave(bool $fallbackToMaster = true): ?Connection
+    public function getSlave(bool $fallbackToMaster = true): ?self
     {
         if (!$this->enableSlaves) {
             return $fallbackToMaster ? $this : null;
         }
-
-        if ($this->slave === null) {
+        if (!$this->slave === null) {
```

Comment:
This change breaks the behavior of the method, and fail tests. i recommend leaving it as it was.

```php
if ($this->slave === null) {
```
```

Reply:
Thanks, that's a typo.